Mage/src/main/java/mage/abilities/common/AttachedToCreatureTriggeredAbility.java

```diff
@@ -0,0 +1,46 @@
+package mage.abilities.common;
+
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.Effect;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+
+import static mage.constants.CardType.CREATURE;
+
+/**
+ *
+ * @author htrajan
+ */
+public class AttachedToCreatureTriggeredAbility extends TriggeredAbilityImpl {
+
+    public AttachedToCreatureTriggeredAbility(Effect effect, boolean optional) {
+        super(Zone.BATTLEFIELD, effect, optional);
+    }
+
+    public AttachedToCreatureTriggeredAbility(final AttachedToCreatureTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.ATTACHED && event.getSourceId().equals(this.getSourceId());
```

Comment:
That's can be wrong code. Check it again, as example from `AuraAttachedTriggeredAbility`: 
* `event.getTargetId().equals(this.getSourceId())`

Also don't use `getSourceId().equals` -- source can be null sometimes.

Reply:
I think the code here is correct because the equipment is firing the trigger, not the attached creature. However, I see your point about source sometimes being null and added a null check prior to checking id equality.